opds_import.py

```diff
@@ -446,8 +448,13 @@ def combine(self, d1, d2):
             return dict(d1)
         new_dict = dict(d1)
         for k, v in d2.items():
-            if k in new_dict and isinstance(v, list):
-                new_dict[k].extend(v)
+            if k in new_dict:
+                if isinstance(v, list):
+                    new_dict[k].extend(v)
+                elif isinstance(v, dict):
+                    new_dict[k] = self.combine(new_dict[k], v)
+                elif new_dict[k] is None:
+                    new_dict[k] = v
```

Comment:
This is where we allow the feedparser code to abstain from providing a value it would normally provide (by returning None). If elementtree provides a value it will override the None value.